nixos/modules/services/misc/sssd.nix

```diff
@@ -0,0 +1,32 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.sssd;
+  sssdConf = pkgs.writeText "sssd/sssd.conf" cfg.sssdConf;
```

Comment:
Cool.  Does sssd.conf typically contain secrets that must not be readable by regular users?  If not, you could probably get away with having nix generate it and fix its permissions at system activation time.